### DISCUSSION TITLE
npm search

### DISCUSSION THREAD

**Discussion by iansu** [johnlindquist/kit#283] [Share]
[Install npm](https://scriptkit.com/api/new?name=npm&url=https://gist.githubusercontent.com/iansu/7f0227f191865e18b5a2b7ff616d6fd9/raw/6e18ce83af9eedc2a3fde3b39f35f682e3d841e4/npm.js)

This is a very simple script that I made to search npm and open the results in your browser. I took a quick look and didn't see anything similar here so I figured I would share it, even though the script itself isn't particularly interesting.

```js
// Menu: npm
// Description: Search npm
// Author: Ian Sutherland
// Twitter: @iansu

await arg("Search query:", async () => {
  let query = await arg("Search query:")

  exec(`open https://www.npmjs.com/search?q=${query}`)
})
```

**Comment by johnlindquist** [comment]
Nice!

You might be interest in the source of the "npm install" script included in Script Kit:

[install.ts](https://github.com/johnlindquist/kit/blob/alpha/src/cli/install.ts)

You can see in action here:

<img width="681" alt="Screen Shot 2021-05-18 at 4 26 03 PM" src="https://user-images.githubusercontent.com/36073/118731019-bf57b980-b7f5-11eb-878e-ad354a74eb11.png">

The packages then render like this:

<img width="669" alt="Screen Shot 2021-05-18 at 4 25 31 PM" src="https://user-images.githubusercontent.com/36073/118730967-acdd8000-b7f5-11eb-8c7e-1b7fe47174a5.png">

**Reply by iansu** [reply]
Nice. I like showing the results in the app like that. Being able to open in a browser is sometimes nice too. Maybe I can figure out how to get both to work.
